src/control/controlpushbutton.cpp

```diff
@@ -10,12 +10,7 @@ ControlPushButton::ControlPushButton(const ConfigKey& key, bool bPersist, double
         : ControlObject(key, false, false, bPersist, defaultValue),
           m_buttonMode(PUSH),
           m_iNoStates(2) {
-    if (m_pControl) {
-        m_pControl->setBehavior(
-                new ControlPushButtonBehavior(
-                        static_cast<ControlPushButtonBehavior::ButtonMode>(m_buttonMode),
-                        m_iNoStates));
-    }
+    setBehavior();
```

Comment:
A setter usually takes a parameter. Maybe `updateBehaviour()`?  

A setBehaviour() function taking num_states and mode would be also beneficial for places like here, where the first behaviour is allocated and destroyed right away: 
https://github.com/mixxxdj/mixxx/blob/8a00cc00e8e5fad1dfaf5d16c1a4bfcd98c60f3e/src/engine/controls/vinylcontrolcontrol.cpp#L42-L43